src/movegen.cpp

```diff
@@ -140,22 +136,21 @@ namespace {
             if (pawnsNotOn7 & dcCandidates)
             {
                 Bitboard dc1 = shift<Up>(pawnsNotOn7 & dcCandidates) & emptySquares & ~file_bb(ksq);
-                Bitboard dc2 = shift<Up>(dc1 & TRank3BB) & emptySquares;
-
-                b1 |= dc1;
-                b2 |= dc2;
+                b |= dc1 | (shift<Up>(dc1 & TRank3BB) & emptySquares);
```

Comment:
How about replacing these two lines with:
```c++
b |= shift<Up>(pawnsNotOn7 & dcCandidates) & emptySquares & ~file_bb(ksq);
b |= shift<Up>(dc1 & TRank3BB) & emptySquares;
```

Reply:
Don't you need to store the dc1 bitboard so that you can correctly shift it Up ?

Reply:
You're right, sorry, I didn't notice that it is used in the second expression.